zipkin-ui/js/component_ui/spanPanel.js

```diff
@@ -44,10 +44,12 @@ export function maybeMarkTransientError(row, anno) {
 export function formatAnnotationValue(value) {
   const type = $.type(value);
   if (type === 'object' || type === 'array' || value == null) {
-    return escapeHtml(JSON.stringify(value));
-  } else {
-    return escapeHtml(value.toString()); // prevents false from coercing to empty!
+    return `<pre><code>${escapeHtml(JSON.stringify(value, null, 2))}</code></pre>`;
   }
+  const result = value.toString();
+  // Preformat if the text includes newlines
+  return result.indexOf('\n') === -1 ? escapeHtml(result)
```

Comment:
unit test both with and without newline to ensure you've done what you want (and it stays like that). Or, remove the special casing of newlines, as annotations should certainly not have newlines

Reply:
I actually copied from the formatBinaryAnnotationValue func, which works fine

Reply:
I agree with your point, let me delete these unnecessary code